src/envoy/http/authn/http_filter.cc

```diff
@@ -14,23 +14,28 @@
  */
 
 #include "src/envoy/http/authn/http_filter.h"
+#include "common/http/utility.h"
 #include "src/envoy/http/authn/mtls_authentication.h"
-#include "src/envoy/utils/utils.h"
 
 namespace Envoy {
 namespace Http {
 
 AuthenticationFilter::AuthenticationFilter(
-    const istio::authentication::v1alpha1::Policy& config)
-    : config_(config) {}
+    const istio::authentication::v1alpha1::Policy& config,
+    Upstream::ClusterManager& cm, IstioAuthn::JwtMultiFactoryStore& jwt_store)
+    : config_(config), cm_(cm), jwt_store_(jwt_store) {}
 
 AuthenticationFilter::~AuthenticationFilter() {}
 
 void AuthenticationFilter::onDestroy() {
   ENVOY_LOG(debug, "Called AuthenticationFilter : {}", __func__);
+  if (jwt_auth_ != nullptr) {
+    jwt_auth_->onDestroy();
+  }
 }
 
-FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap&, bool) {
+FilterHeadersStatus AuthenticationFilter::decodeHeaders(HeaderMap& headers,
+                                                        bool) {
```

Comment:
Please rebase so you can hook it up to the proper place.

Reply:
Will rebase.

Reply:
Done.